code/script.js

```diff
@@ -253,91 +256,130 @@ const setSecret = () => {
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  generateBoard() 
+  setSecret()  
+  console.log(secret)
+  board.style.display = 'flex'
+  winOrLose.style.display = 'none'
+
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  const chosenValue = questions.value
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value: chosenValue,
+      category: category,      
     }
   } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: 'eyeColor',
+      value: chosenValue,
+      category: category,
+    }
+  } else if (category === 'accessories' || category === 'other') { //used || instead of creating 2 else if's
+    currentQuestion = {
+      attribute: chosenValue,
+      value: true, 
       category: category,
     }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
+// Invoked when FindOutButton clicked.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-}
+  let value = currentQuestion.value
+  let attribute = currentQuestion.attribute
+  if (value === secret[attribute]) {
+      filterCharacters(true)
+  } else {
+      filterCharacters(false)
+  }
+
+ }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  let category = currentQuestion.attribute
+   if (category === 'accessories') {
```

Comment:
Great decision to reuse 'category' here, and remove the prewritten group. It makes it easier to follow the code!

Reply:
By changing currentQuestion.attribute to currentQuestion.category, all my alerts started to function. :)